.github/workflows/ci-upstream-dev.yml

```
@@ -0,0 +1,47 @@
+name: CI upstream development versions
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+  workflow_dispatch: # allows you to trigger the workflow run manually
+  schedule:
+    - cron: "5 2 * * tue" # Tuesdays at 02:05 UTC.
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+        python-version: [ "3.12" ]
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up uv and caching
+      uses: astral-sh/setup-uv@v2
+      with:
+        enable-cache: true
+        cache-suffix: "${{ matrix.os }}-${{ matrix.python-version }}-upstream-dev"
+        cache-dependency-glob: "setup.cfg"
+        version: "0.4.9"
+
+    - name: Create venv
+      run: |
+        uv venv --seed --python ${{ matrix.python-version }}
+
+    - name: Install ${{ matrix.resolution }} dependencies
+      run: |
+        uv pip install --resolution highest --prerelease allow .[test,upstream-dev]
+
+    - name: Test with pytest (linux)
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        source .venv/bin/activate
+        pytest --xdoc
```

```suggestion
        uv run pytest --xdoc
```

Or does this not work for a reason I can't see?

---

I think uv is installed in the venv, so things are somewhat circular. But maybe not, I'll test!

---

No, it fails because we haven't fully adopted uv yet, just surgically replaced `pip` with `uv pip` and `venv` with `uv venv`. I tried fully adopting uv when I did that, but building was hard or something, after messing around for half an hour I decided to postpone that for a few more months until uv can do everything.

See the failing CI here: https://github.com/primap-community/primap2/actions/runs/11445079855/job/31841383315?pr=275

---

Ah ok fair, makes sense. Thanks!